Automattic/sensei
# Add Github Actions workflow to build plugin on each PR

### Changes proposed in this Pull Request

* Installs all the dependencies, builds the plugin, and uploads it as an artifact to be stored for 7 days.
   * This will be complemented by PR #4884, and it's a follow-up of PR #4731.


### Testing instructions

Check if the check "Plugin Build" below ran successfully, and if it has an artifact called `sensei-33c0d28f03d980caa0a58b26b0a77e49d497f950` (or whatever the last commit on this PR is) that, when downloaded, becomes  `sensei-33c0d28f03d980caa0a58b26b0a77e49d497f950.zip` (please note the extension `.zip`) and is a build of the plugin as generated by our usual release process.
